# SARST2 v2.0.26
 Structural similarity search Aided by Ramachandran Sequential Transformation, version 2

<p align="center">
  <img width="512" src="https://github.com/NYCU-10lab/sarst/blob/main/Manuals/SARST2-logo.png"><br>
  <img width="300" src="https://github.com/user-attachments/assets/5e1fac98-fa11-4dff-8408-4a863d50d0ac">
</p>

## About this software
SARST2 (Structural similarity search Aided by Ramachandran Sequential Transformation, version 2) is a high-performance protein structure alignment algorithm. It supports both database searches for structural similarity using a given query protein, as well as pairwise structure alignments between two protein structures.

This software is published along with the following paper and will be frequently updated via the URLs provided in the publication:

<table>
  <tr>
    <td><b>Title</b></td>
    <td>SARST2, a high-throughput protein structure alignment algorithm for searching massive databases</td>
  </tr>
  <tr>
    <td><b>Authors</b></td>
    <td>Wei-Cheng Lo*, Arieh Warshel, Chia-Hua Lo, Chia Yee Choke, Yan-Jie Li, Shih-Chung Yen, Jyun-Yi Yang and Shih-Wen Weng</td>
  </tr>
  <tr>
    <td><b>Institute</b></td>
    <td>Institute of Bioinformatics and Systems Biology, National Yang Ming Chiao Tung University, Hsinchu, Taiwan, Republic of China</td>
  </tr>
</table>
*Corresponding author (WadeLo@nycu.edu.tw)<br><br>
Download URLs:<br>
https://github.com/NYCU-10lab/sarst<br>
https://10lab.ceb.nycu.edu.tw/sarst2

## Download
<table border="1">
  <tr> 
    <td width="200" valign="top" nowrap> 
      <b>Standalone program</b>
    </td>
    <td width="380" nowrap>
      <table border="1" cellpadding="0" cellspacing="0" bordercolor="#DDDDDD">
        <tr>
          <td>Linux, x86_64</td>
          <td nowrap>
            <a target="_blank" href="https://10lab.ceb.nycu.edu.tw/sarst2/dat/downloads/prgs/SARST2-v2.0.26-Linux.x86_64.tar.gz">
              .tar.gz 
            </a>
            &nbsp;&nbsp;
            <a target="_blank" href="https://10lab.ceb.nycu.edu.tw/sarst2/dat/downloads/prgs/SARST2-v2.0.26-Linux.x86_64.zip">
              .zip 
            </a>
          </td>
        </tr>
        <tr>
          <td>Linux, arm64</td>
          <td nowrap>
            <a target="_blank" href="https://10lab.ceb.nycu.edu.tw/sarst2/dat/downloads/prgs/SARST2-v2.0.26-Linux.arm64.tar.gz">
              .tar.gz
            </a>
            &nbsp;&nbsp;
            <a target="_blank" href="https://10lab.ceb.nycu.edu.tw/sarst2/dat/downloads/prgs/SARST2-v2.0.26-Linux.arm64.zip">
              .zip
            </a>
          </td>
        </tr>
        <tr>
          <td>macOS, x86_64</td>
          <td nowrap>
            <a target="_blank" href="https://10lab.ceb.nycu.edu.tw/sarst2/dat/downloads/prgs/SARST2-v2.0.26-macOS.x86_64.tar.gz">
              .tar.gz
            </a>
            &nbsp;&nbsp;
            <a target="_blank" href="https://10lab.ceb.nycu.edu.tw/sarst2/dat/downloads/prgs/SARST2-v2.0.26-macOS.x86_64.zip"">
              .zip
            </a>
          </td>
        </tr>
        <tr>
          <td>macOS, arm64</td>
          <td nowrap>
            <a target="_blank" href="https://10lab.ceb.nycu.edu.tw/sarst2/dat/downloads/prgs/SARST2-v2.0.26-macOS.arm64.tar.gz">
              .tar.gz
            </a>
            &nbsp;&nbsp;
            <a target="_blank" href="https://10lab.ceb.nycu.edu.tw/sarst2/dat/downloads/prgs/SARST2-v2.0.26-macOS.arm64.zip">
              .zip
            </a>
          </td>
        </tr>
        <tr>
          <td nowrap>Windows 10+, x86_64</td>
          <td nowrap>
            <a target="_blank" href="https://10lab.ceb.nycu.edu.tw/sarst2/dat/downloads/prgs/SARST2-v2.0.26-Win10+.x86_64.tar.gz">
              .tar.gz
            </a>
            &nbsp;&nbsp;
            <a target="_blank" href="https://10lab.ceb.nycu.edu.tw/sarst2/dat/downloads/prgs/SARST2-v2.0.26-Win10+.x86_64.zip">
              .zip
            </a>
          </td>
        </tr>
        <tr>
          <td>Windows 10+, arm64</td>
          <td nowrap>
            <a target="_blank" href="https://10lab.ceb.nycu.edu.tw/sarst2/dat/downloads/prgs/SARST2-v2.0.26-Win10+.arm64.tar.gz">
              .tar.gz
            </a>
            &nbsp;&nbsp;
            <a target="_blank" href="https://10lab.ceb.nycu.edu.tw/sarst2/dat/downloads/prgs/SARST2-v2.0.26-Win10+.arm64.zip">
              .zip
            </a>
          </td>
        </tr>
      </table>
    </td>
  </tr>
</table>

<br>

<table border="1" cellpadding="0" cellspacing="0">
  <tr> 
    <td width="200" valign="top" nowrap>
      <b>Target database</b>
    </td>
    <td width="380" nowrap>
      <b>PDB-2022</b><br>
      <Table border="1" cellpadding="0" cellspacing="0" bordercolor="#DDDDDD",>
        <tr>
          <td width="270" nowrap>
            &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
            <a target="_blank" href="https://10lab.ceb.nycu.edu.tw/sarst2/dat/downloads/tdbs/PDB-2022.tar.gz">
              .tar.gz
            </a>
            &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
            <a target="_blank" href="https://10lab.ceb.nycu.edu.tw/sarst2/dat/downloads/tdbs/PDB-2022.zip">
              .zip
            </a>
          </td>
        </tr>
        <tr>
          <td width="270" nowrap>
            &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
            GoogleDrive:
            &nbsp;&nbsp;
            <a target="_blank" href="https://drive.google.com/file/d/1xUiHe6ov7lnMKS8NDgn0o1goP_n0Qr2s/view?usp=sharing">
              .tar.gz
            </a>
            &nbsp;&nbsp;
            <a target="_blank" href="https://drive.google.com/file/d/1q9xjsI9KPlQZl1BEOEK8Py8AqymFl62M/view?usp=sharing">
              .zip
            </a>
          </td>
        </tr>
      </Table>
      <br>
      <b>SCOP-2.07</b><br>
      <Table border="1" cellpadding="0" cellspacing="0" bordercolor="#DDDDDD",>
        <tr>
          <td width="270" nowrap>
            &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
            <a target="_blank" href="https://10lab.ceb.nycu.edu.tw/sarst2/dat/downloads/tdbs/SCOP-2.07.tar.gz">
              .tar.gz
            </a>
            &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
            <a target="_blank" href="https://10lab.ceb.nycu.edu.tw/sarst2/dat/downloads/tdbs/SCOP-2.07.zip">
              .zip
            </a>
          </td>
        </tr>
        <tr>
          <td width="270" nowrap>
            &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
            GoogleDrive:
            &nbsp;&nbsp;
            <a target="_blank" href="https://drive.google.com/file/d/1JJY4z6HLu9j_q6CqNBIJxAvwGqfcxf8l/view?usp=sharing">
              .tar.gz
            </a>
            &nbsp;&nbsp;
            <a target="_blank" href="https://drive.google.com/file/d/1FM2rcVGrNg1qpTn0xhjGLaDKviZXGTru/view?usp=sharing">
              .zip
            </a>
          </td>
        </tr>
      </Table>
      <br>
      <b>Qry200 & Qry400 datasets</b><br>
      <Table border="1" cellpadding="0" cellspacing="0" bordercolor="#DDDDDD",>
        <tr>
          <td width="270" nowrap>
            &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
            <a target="_blank" href="https://10lab.ceb.nycu.edu.tw/sarst2/dat/downloads/qrys/QuerySets.tar.gz">
              .tar.gz
            </a>
            &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
            <a target="_blank" href="https://10lab.ceb.nycu.edu.tw/sarst2/dat/downloads/qrys/QuerySets.zip">
              .zip
            </a>
          </td>
        </tr>
      </Table>
      <br>
      <b>AFDB-2022</b><br>
      <Table border="1" cellpadding="0" cellspacing="0" bordercolor="#DDDDDD",>
        <tr>
          <td width="270" nowrap>
            &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
            <a target="_blank" href="https://10lab.ceb.nycu.edu.tw/sarst2/dat/downloads/tdbs/AFDB/AFDB-2022.tar.gz">
              .tar.gz
            </a>
          </td>
        </tr>
      </Table>
      <br>
    </td>
  </tr>
</table>

## User manual

| Language (alphabetical order) | Link         |
| ------------------- |:------------:|
| Chinese | [繁体中文](https://github.com/NYCU-10lab/sarst/blob/main/Manuals/SARST2-UserManual-Chinese%2CTraditional.pdf)<br>[简体中文](https://github.com/NYCU-10lab/sarst/blob/main/Manuals/SARST2-UserManual-Chinese%2CSimplified.pdf) |
| English             | [English](https://github.com/NYCU-10lab/sarst/blob/main/Manuals/SARST2-UserManual-English.pdf) |
| French              | [Français](https://github.com/NYCU-10lab/sarst/blob/main/Manuals/SARST2-UserManual-French.pdf) |
| German              | [Deutsch](https://github.com/NYCU-10lab/sarst/blob/main/Manuals/SARST2-UserManual-German.pdf) |
| Japanese            | [日本語](https://github.com/NYCU-10lab/sarst/blob/main/Manuals/SARST2-UserManual-Japanese.pdf) |
| Korean              | [한국어](https://github.com/NYCU-10lab/sarst/blob/main/Manuals/SARST2-UserManual-Korean.pdf) |

<br>

### 1.	Installation
The latest version of the SARST2 program and pre-formatted target databases are available at the URLs listed above.
After downloading a compressed archive, use tar, gzip, or zip utilities to extract the files, depending on the format of the archive. Once extracted, the SARST2 program files are provided as pre-compiled executable binaries and do not require installation.

For example, if you download the archive SARST2-v2.0.26-Linux.x86_64.tar.gz, you can decompress it under Linux using the following command: <br>
```tar xfp SARST2-v2.0.26-Linux.x86_64.tar.gz```

After extraction, you can run the sarst2 program with the following commands:<br>
```cd SARST2-v2.0.26-Linux.x86_64/bin```<br>
```chmod +x sarst2```<br>
```./sarst2 -h```

### 2.	Contents of the software folders
<table>
  <tr>
    <td>bin</td>
    <td>64-bit executables for Linux, Windows, and macOS.</td>
  </tr>
  <tr>
    <td>dbs</td>
    <td>Pre-formatted target databases: PDB-2022 and SCOP-2.07.</td>
  </tr>
  <tr>
    <td>doc</td>
    <td>User manuals in multiple languages.</td>
  </tr>
</table>

### 3.	Quick guide
There are three main programs in this software package, as described below,
<table>
  <tr>
    <td>sarst2</td>
    <td>Implementation of the SARST2 protein structure alignment algorithm.</td>
  </tr>
  <tr>
    <td>formatdb</td>
    <td>A database formatting tool that allows users to create custom target databases for structure search and alignment.</td>
  </tr>
  <tr>
    <td>readdb</td>
    <td>A tool for extracting the protein amino acid and linearly-encoded structure sequences stored in a pre-formatted target database.</td>
  </tr>
</table>

Running any of the above programs without parameters (or with -h) will display a brief text-based help message for that program.

#### Linux, macOS
- ```./sarst2```
- ```./formatdb```
- ```./readdb```
- ```./sarst2 -h```
- ```./formatdb -h```
- ```./readdb -h```

#### Windows (Cmd or PowerShell)
- ```.\sarst2.exe```
- ```.\formatdb.exe```
- ```.\readdb.exe```
- ```.\sarst2.exe -h```
- ```.\formatdb.exe -h```
- ```.\readdb.exe -h```  

### 4.	Manual: sarst2
#### 4.1	Usage
```
./sarst2	query structure		subject structure(s)	[Options]
               ---------------		-------------------
               > one PDB/CIF file	> can be
                                        1. -db + a pre-formatted database
                                        2. multiple PDB/CIF files
                                        3. folders of PDB/CIF files
                                        4. one PDB/CIF file (pairwise)
```

#### 4.2	Program options
<table>
<tr>
    <td>-db</td>
    <td>[str]</td>
    <td>The target database of subject structures to search.<br>(default: none)</td>
</tr>
<tr>
    <td>-brief</td>
    <td>[int]</td>
    <td>Number of subject structures to show one-line summaries.<br>(default: 500)</td>
</tr>
<tr>
    <td>-detail</td>
    <td>[int]</td>
    <td>Number of subject structures to show detailed alignment data.<br>(default: 500)</td>
</tr>
<tr>
    <td>-t</td>
    <td>[int]</td>
    <td>Number of threads.<br>It must be ≥ 0; 0 means all processors will be used.<br>(default: 0, all processors)</td>
</tr>
<tr>
    <td>-w</td>
    <td>[int]</td>
    <td>Word size.<br>(default: 5)</td>
</tr>
<tr>
    <td>-orderby</td>
    <td>[int]</td>
    <td>Sort the hit list by one of the following factors, 1: Conf-score--, 2: TM-score--, 3: sequence identity--, or 4: RMSD++, where --/++ means descending/ascending order.<br>(default: 1, Conf-score--)<br>(ignored in a pairwise alignment)</td>
</tr>
<tr>
    <td>-mode</td>
    <td>[int]</td>
    <td>Searching mode, 1: accurate, 2: balanced, 3: quick, or other values: auto.<br>(default: auto)</td>
</tr>
<tr>
    <td>-f</td>
    <td>[int]</td>
    <td>Enable minor filters, 0: off, 1: on, or other values: auto.<br>(default: auto)<br>(always 0, disabled, in a pairwise alignment)</td>
</tr>
<tr>
    <td>-C</td>
    <td>[float]</td>
    <td>Conf-score (confidence score) threshold.<br>It must be between 0 and 1; 0 means no threshold is applied.<br>(default: 0.5)<br>(always disabled in a pairwise alignment)</td>
</tr>
<tr>
    <td>-pC</td>
    <td>[float]</td>
    <td>Cutoff of the final pC-value, i.e., –log2(C).<br>It must be ≥ 0; 0 means no cutoff is applied.<br>(default: 1.0, equivalent to -C = 0.5)<br>(always disabled in a pairwise alignment)</td>
</tr>
<tr>
    <td>-e</td>
    <td>[float]</td>
    <td>Cutoff of the pC-value, applied to each filter and refinement step.<br>Given the same -e and -pC, -e discards more irrelevant hits.<br>It must be ≥ 0; 0 means no cutoff is applied.<br>(default: 1.0)<br>(always disabled in a pairwise alignment)</td>
</tr>
<tr>
    <td>-tmcut</td>
    <td>[float]</td>
    <td>TM-score cutoff.<br>It must be ≥ 0; 0 means no cutoff is applied.<br>A TM-score ≥ 0.7 by SARST2 might imply family-level homology.<br>(default: 0.15)<br>(always disabled in a pairwise alignment)
</td>
</tr>
<tr>
    <td>-mem</td>
    <td>[T/F]</td>
    <td>Cache all subject protein data in memory.<br>(default: T)<br>(always T, enabled, in a pairwise alignment)</td>
</tr>
<tr>
    <td>-q</td>
    <td>[T/F]</td>
    <td>Quick output style.<br>Display results in a simplified, parser-friendly format.<br>(default: F)</td>
</tr>
<tr>
    <td>-sa</td>
    <td>[T/F]</td>
    <td>Display the structure-based sequence alignment.<br>(default: T)</td>
</tr>
<tr>
    <td>-mat</td>
    <td>[T/F]</td>
    <td>Display the transformation matrix for superimposition.<br>(default: F)</td>
</tr>
<tr>
    <td>-nmsbj</td>
    <td>[T/F]</td>
    <td>Normalize the TM-score by the size of the subject structure.<br>(default: F)</td>
</tr>
<tr>
    <td>-nmavg</td>
    <td>[T/F]</td>
    <td>Normalize the TM-score by the average size of the query structure and each subject structure.<br>(default: F)</td>
</tr>
<tr>
    <td>-nmusr</td>
    <td>[float]</td>
    <td>The protein size for normalizing the TM-score.<br>It should be >= minimum size of the two structures; otherwise, the TM-score may be > 1.</td>
</tr>
<tr>
    <td>-d</td>
    <td>[float]</td>
    <td>The d0 for scaling the TM-score, e.g., 5.0 Angstroms (Å).</td>
</tr>
<tr>
    <td>-ml</td>
    <td>[T/F]</td>
    <td>Apply machine learning.<br>(default: T)<br>(always F, disabled, in a pairwise alignment)</td>
</tr>
<tr>
    <td>-fdp</td>
    <td>[str]</td>
    <td>Dynamic programming algorithm for the filtering steps.<br>Supported options: NW (Needleman-Wunsch), SW (Smith-Waterman)<br>(default: NW)</td>
</tr>
<tr>
    <td>-rdp</td>
    <td>[str]</td>
    <td>Dynamic programming algorithm for the refinement step.<br>Supported options: NW (Needleman-Wunsch), SW (Smith-Waterman)<br>(default: NW)</td>
</tr>
<tr>
    <td>-swp</td>
    <td>[str]</td>
    <td>Path to the user-specified swap file.<br>Using a swap file can reduce the memory cost.<br>(default: none)</td>
</tr>
<tr>
    <td>-Sout</td>
    <td>[str]</td>
    <td>Folder to output the structure superimposition files.<br>The folder will be created if it does not exist.<br>The number of superimposition files is confined by the -detail option.<br>(default: none)</td>
</tr>
<tr>
    <td>-html</td>
    <td>[str]</td>
    <td>Make an HTML output folder.<br>Superimposed structure files will also be generated in the HTML folder.<br>(default: none)</td>
</tr>
<tr>
    <td>-jsmol</td>
    <td>[str]</td>
    <td>Set the path to the JSmol JavaScript package for displaying superimposed structures in the HTML output.<br>It can be a local disk folder or an HTTP(S) URL.<br>(default: none)<br>(trial URL: "https://10lab.ceb.nycu.edu.tw/ext/jsmol")</td>
</tr>
<tr>
    <td>-pssm_out</td>
    <td>[str]</td>
    <td>File to store the PSSMs of the structural and sequence codes applied in this algorithm.<br>(default: none)</td>
</tr>
<tr>
    <td>-pssm_pC</td>
    <td>[float]</td>
    <td>The pC-value cutoff for PSSM construction.<br>(default: 0.05)</td>
</tr>
<tr>
    <td>-h</td>
    <td></td>
    <td>Print the help message (quick guide).</td>
</tr>
</table>

#### 4.3	Example usage: database structural similarity search
##### Search the query structure against a pre-formatted target database 
```./sarst2 Qry.pdb -db my_db/my_proteins.db -brief 10 -w 7 -e 0.1```  
```./sarst2 Qry.pdb -db my_db/my_proteins.db -brief 10 -d 5.0 -sa F```  

In the above example, the target database is stored in the folder "my_db", and "my_proteins.db" is the file stem of the target database files. See the [**Manual**: **formatdb**] for the actual database file list.

#### 4.4	Example usage: one-against-all alignments
##### Search the query structure against listed subject structures
```./sarst2 Qry.pdb Sbj1.pdb Sbj2.cif Sbj3.pdb -mat T```  

##### Search the query structure against subject structure files specified with wildcard patterns 
```./sarst2 Qry.pdb "set1/*.pdb" "set2/1a???.cif" -nmavg T```  

In this example, "set1/*.pdb" and "set2/1a???.cif" are enclosed in quotation marks and contain wildcard characters. The sarst2 program will automatically expand these wildcard patterns and retrieve the matching file names internally. If the patterns are not enclosed in quotation marks, the operating system will expand the wildcards instead. When the number of matching files is large, the resulting command-line argument list may exceed system limits and cause the command to fail. Therefore, we recommend enclosing wildcard patterns in quotation marks to allow sarst2 to handle file listing internally, rather than relying on the operating system's default behavior.

##### Search the query structure against several folders containing subject structures 
```./sarst2 Qry.pdb set1 set2 -nmavg T```  

In this example, set1 and set2 are folders that may contain protein structure files. The sarst2 program will automatically retrieve all files in these folders (equivalent to set1/\* and set2/\*). Files identified as PDB or CIF format will be selected and aligned against the query structure.

#### 4.5	Example usage: pairwise structure alignment
##### Align the query structure with one subject structure 
```./sarst2 Qry.pdb Sbj.pdb -sa F```  
```./sarst2 Qry.pdb Sbj.cif -mat T```  

#### 4.6	Output protein structural superimpositions
##### Generate query-subject structural superimposition PDB files 
```./sarst2 Qry.pdb -db prot/myDb –detail 100 –Sout output_folder```  
```./sarst2 Qry.pdb "set1/*.cif" –detail 100 –Sout output_folder```  

Using the -Sout output_folder option, the superimposed protein structures in PDB format will be output to the user-specified folder. The number of superimposed structures generated is defined by the -detail option. Each output file is named Qry-SbjSN.pdb, where SN represents the serial number of the subject protein in the hit list. In each superimposition file, the chain IDs for the query and subject protein structures are Q and S, respectively. The two chains are separated by a TER record, as illustrated below:

![image1](https://github.com/NYCU-10lab/sarst/blob/main/Manuals/Fig_1.png)

As shown in the figure, only alpha carbon (Cα) atoms appear in the superimposition file. This is because SARST2 performs all computations based solely on the Cα coordinates. The orientation of the query structure remains fixed across all superimposition files, while each subject structure is transformed (rotated and translated) according to its alignment with the query structure to achieve superimposition. 

To visualize the superimposed structures, we recommend using RasMol or RasWin (http://www.openrasmol.org/). Since only Cα atoms are present in the superimposition files, the display mode in RasMol should be set to "backbone".

#### 4.7	Generate interactive HTML result pages
##### Generate an HTML document with online JSmol scripts
```./sarst2 Qry.pdb -db my_db/my_proteins.db –html output_folder –jsmol "https://10lab.ceb.nycu.edu.tw/ext/jsmol"``` 

##### Generate an HTML document with a local JSmol script folder (Windows)
```./sarst2 Qry.pdb "set1/*.cif" –detail 100 –html output_folder –jsmol "file:///D:/bioinfo/software/jsmol"```  

Using the "-html output_folder" option, an HTML result document and corresponding superimposed protein structure files will be generated in the specified folder. The 
"-html" option must be used together with "-jsmol", which specifies the URL or local path of the JSmol package (version 2013). JSmol is an interactive 3D molecular structure viewer that runs in web browsers and supports most major modern browsers.

![image2](https://github.com/NYCU-10lab/sarst/blob/main/Manuals/Fig_2.png)

The main file in the HTML output folder is SarstResults.html, which should be opened in a web browser. Other HTML files are embedded within the main page using inner frames. A subfolder named "sup" will also be created; it stores the structural superimposition files between the query and each subject protein in the hit list.

Depending on your operating system, browser, or antivirus software, you may need to adjust the security settings to allow the browser to execute JavaScript and access the superimposed structure files in the "sup" folder, so that the JSmol 3D viewer can function properly.

### 5.	Manual: formatdb
#### 5.1	Usage
```
./formatdb	subject structure(s)	-db database	[Options]
                --------------------
                > can be
                1. multiple PDB/CIF files
                2. folders containing PDB/CIF files
                3. a plain text file listing PDB/CIF file paths
```

#### 5.2	Program options
<table>
<tr>
    <td>-db</td>
    <td>[str]</td>
    <td>The target database of subject structures to create.<br>(default: none)</td>
</tr>
<tr>
    <td>-flist</td>
    <td>[str]</td>
    <td>Plain text file listing PDB/CIF file paths. This argument can be used along with common subject file arguments.<br>(default: none)<br></td>
</tr>
<tr>
    <td>-t</td>
    <td>[int]</td>
    <td>Number of threads.<br>It must be ≥ 0; 0 means all processors will be used.<br>(default: 0, all processors)<br></td>
</tr>
<tr>
    <td>-split</td>
    <td>[int]</td>
    <td>Split the database into subsets, each with the number of subject structures specified by this option. Database splitting helps prevent the database files from exceeding the file size limit of the disk.<br>(default: none)<br></td>
</tr>
<tr>
    <td>-save_disk</td>
    <td>[T/F]</td>
    <td>Round the coordinates of atoms from three decimal places into one decimal place to reduce disk usage.<br>(default: F)</td>
</tr>
<tr>
    <td>-keep_order</td>
    <td>[T/F]</td>
    <td>Keep the order of the subject structures stored in the database as their input order. Setting T may slow down database creation.<br>(default: F)</td>
</tr>
<tr>
    <td>-h</td>
    <td></td>
    <td>Print the help message (quick guide).</td>
</tr>
</table>

#### 5.3	Example usages
##### Create a target database for listed subject structure files
```./formatdb Sbj1.pdb Sbj2.cif Sbj3.pdb -db myDb -keep_order T```  

Several database files with filenames starting with myDb will be created. Enabling -keep_order will preserve the order of subject structures in the target database according to how they were listed in the command-line arguments.

##### Create a target database for listed subject structure files with wildcards
```./formatdb "set1/*.pdb" "set2/*.cif" Sbj1.pdb Sbj2.cif -db myDb```  

When listing subject files, it is fine to mix arguments with and without wildcards. It is recommended to enclose wildcard arguments in quotation marks, so that the program can correctly handle file expansion.

##### Create a target database based on a file list of subject structures
```./formatdb -flist protlist.txt Sbj1.pdb Sbj2.cif -db myDb```

The file protlist.txt should contain a list of file paths, with one file path per line.

##### Create a target database for folders containing subject structure files
```./formatdb folder1 folder2 -db myDb -save_disk T -split 50000```  

Enabling -save_disk will round the Cα coordinates to one decimal place to save storage space. The "-split 50000" option will result in multiple subset databases, each containing at most 50,000 structures. This split option is particularly useful when the size of the formatted database files may exceed the maximum file size supported by some operating systems or disk formats.

### 6.	Manual: readdb
#### 6.1	Usage
```
./readdb    target database     output file     [-seq sequence type]
            ---------------     -----------           -------------
            > must be SARST2    > will be in          > can be
              pre-formatted       FASTA format        1. AA
                                                      2. AAT
                                                      3. SARST
                                                      4. SSE
```

#### 6.2	Program options
<table>
<tr>
    <td>-seq</td>
    <td>[str]</td>
    <td>The output sequence type.<br>AA		amino acid sequence<br>AAT		five-symbol amino acid type sequence<br>SARST	SARST Ramachandran-code sequence<br>SSE		four-symbol secondary structure element sequence<br>(default: AA)</td>
</tr>
<tr>
    <td>-h</td>
    <td></td>
    <td>Print the help message (quick guide).</td>
</tr>
</table>

#### 6.3	Example usages
##### Extract subject sequences from a SARST2 target database 
```./readdb my_db/my_proteins.db seqs.fasta```  
```./readdb my_db/my_proteins.db seqs.fasta -seq SARST```  
```./readdb my_db/my_proteins.db seqs.fasta -seq AAT```  

When -seq is not specified, the default output sequence type is amino acid sequences. The output sequence file will be in FASTA format. If the output file already exists before running readdb, it will be overwritten.
